fedimint-server/src/net/api.rs

```diff
@@ -119,46 +124,14 @@ impl ConsensusApi {
         // We ignore any writes, as we only verify if the transaction is valid here
         dbtx.ignore_uncommitted();
 
-        let mut funding_verifier = FundingVerifier::default();
-        let mut public_keys = Vec::new();
-
-        for input in transaction.inputs.iter() {
-            let meta = self
-                .modules
-                .get_expect(input.module_instance_id())
-                .process_input(
-                    &mut dbtx.dbtx_ref_with_prefix_module_id(input.module_instance_id()),
-                    input,
-                )
-                .await?;
-
-            funding_verifier.add_input(meta.amount);
-            public_keys.push(meta.pub_keys);
-        }
-
-        transaction.validate_signature(public_keys.into_iter().flatten())?;
-
-        for (output, out_idx) in transaction.outputs.iter().zip(0u64..) {
-            let amount = self
-                .modules
-                .get_expect(output.module_instance_id())
-                .process_output(
-                    &mut dbtx.dbtx_ref_with_prefix_module_id(output.module_instance_id()),
-                    output,
-                    OutPoint { txid, out_idx },
-                )
-                .await?;
-
-            funding_verifier.add_output(amount);
-        }
-
-        funding_verifier.verify_funding()?;
+        process_transaction_with_dbtx(self.modules.clone(), &mut dbtx, transaction.clone()).await?;
 
         self.submission_sender
             .send(ConsensusItem::Transaction(transaction))
-            .await?;
+            .await
+            .ok();
 
```

Comment:
Should we log if that ever fails? Or can this normally happen during shutdown?

Reply:
May happen during shutdown I believe.